### DISCUSSION TITLE
how to listen for MouseWheel on a egui Image ?

### DISCUSSION THREAD

**Discussion by m172021** [emilk/egui#4531] [General]
Using code outside of Egui, I am rendering a "map" to a Texture, which I then display in Egui as an Image.

I can display the Image right now. I can also listen for drag events on the img via:

```
let resp = ui.add(img.sense(Sense::drag()));
```

this is enough for panning left/right up/down

I also want to intercept MouseWheel events on the image (so I can call the underlying custom wgpu map rendering code to zoom in/out).

How do I listen to MouseWheel events on this egui Image ?

Thanks!

**Comment by YgorSouza** [comment]
This isn't available as a method in the `Response` like `hovered()` and `clicked()`, but you can get the scroll wheel events from `ui.input()` and use them when the image response is hovered. Here is a sample based on the hello_world example:

<details><summary>Click to open example</summary>
<p>

```rs
struct MyApp {
    age: u32,
}

impl Default for MyApp {
    fn default() -> Self {
        Self { age: 42 }
    }
}

impl eframe::App for MyApp {
    fn update(&mut self, ctx: &egui::Context, _frame: &mut eframe::Frame) {
        egui::CentralPanel::default().show(ctx, |ui| {
            ui.label(format!("N = {}", self.age));
            let response = ui.image(egui::include_image!(
                "../../../crates/egui/assets/ferris.png"
            ));
            if response.hovered() {
                let delta = ui.input(|i| {
                    i.events.iter().find_map(|e| match e {
                        egui::Event::MouseWheel {
                            unit: _,
                            delta,
                            modifiers,
                        } if modifiers.command_only() => Some(*delta),
                        _ => None,
                    })
                });
                if let Some(delta) = delta {
                    self.age = self
                        .age
                        .wrapping_add_signed(delta.max_elem().signum() as i32);
                }
            }
        });
    }
}
```


https://github.com/emilk/egui/assets/43298013/eb7442e5-b711-4fd7-843b-976432be2462

</p>
</details> 


If you just want to use the zoom and scroll events as defined by egui (e.g., Ctrl to zoom, Shift to scroll horizontally), you can also do what the plot widget does:

https://github.com/emilk/egui/blob/a98c42e3177fa31807c37921a0d38734e2f7597f/crates/egui_plot/src/lib.rs#L1093-L1130
